lib/connectors_async/job_runner.rb

```diff
@@ -63,16 +64,15 @@ def log_with_thread_id(level, str)
     private
 
     def with_throttling(job)
-      attempts = 0
       if job.should_wait?
         log_with_thread_id(:info, "Job #{job.id} is sleeping: Enterprise Search hasn't picked up documents for a while.")
-
+        wait_duration = 0
         while job.should_wait?
-          if attempts < ConnectorsShared::Constants::MAX_IDLE_ATTEMPTS
-            attempts += 1
+          if wait_duration < ConnectorsShared::Constants::MAX_IDLE_TIME
```

Comment:
Do we want to be precise about it?

E.g. if setting MAX_IDLE_TIME = 10.seconds and IDLE_SLEEP_TIME = 3.seconds, we will actually sleep for 12.seconds.

Usability-wise it's not a big problem, yet something that might not be immediately expected.